core/src/main/scala/org/apache/spark/scheduler/cluster/StandaloneSchedulerBackend.scala

```diff
@@ -111,15 +112,19 @@ private[spark] class StandaloneSchedulerBackend(
     // ExecutorAllocationManager will send the real initial limit to the Master later.
     val initialExecutorLimit =
       if (Utils.isDynamicAllocationEnabled(conf)) {
+        if (coresPerExecutor.isEmpty) {
+          logWarning("Dynamic allocation without explicitly setting spark.executor.cores " +
+            "detected, you may get more executors allocated than expected. It's recommended to " +
```

Comment:
nit: "Dynamic allocation enabled without spark.executor.cores explicitly set, you may ..."